pkgs/games/runelite/default.nix

```diff
@@ -1,53 +1,65 @@
-{ stdenv, fetchurl, makeWrapper, jre, makeDesktopItem, lib }:
+{ stdenv, lib, pkgs, ... }:
 
 stdenv.mkDerivation rec {
   pname = "runelite";
-  version = "2.0.0";
+  version = "2.1.5";
 
-  src = fetchurl {
+  jar = pkgs.fetchurl {
     url = "https://github.com/runelite/launcher/releases/download/${version}/RuneLite.jar";
-    sha256 = "0q0x8g43ggkvp5fgnymgygx59xfhnyyrhpi6ha333gqg97rr0rvw";
+    sha256 = "4BX188QIjIFTxng2ktqlKn7AqQ9tdBcKWmgOj/5yd10=";
   };
 
-  icon = fetchurl {
+  icon = pkgs.fetchurl {
     url = "https://github.com/runelite/launcher/raw/${version}/appimage/runelite.png";
     sha256 = "04fcjm7p546gr82g0jbh497j7rnh70lrvas0k171bff4v3knrjw1";
   };
 
-  desktop = makeDesktopItem {
-    name = "RuneLite";
-    type = "Application";
-    exec = "runelite";
-    icon = icon;
-    comment = "Open source Old School RuneScape client";
-    terminal = "false";
-    desktopName = "RuneLite";
-    genericName = "Oldschool Runescape";
-    categories = "Game";
-    startupNotify = null;
+  # The `.so` files provided by these two jars aren't detected by RuneLite for some reason, so we have to provide them manually
```

Comment:
Are these distributed as JARs? what could be happening is that nix isn't aware of them, and will "filter them out" as part of the fixup phases

Reply:
The JARs are not distributed with runelite itself, they are downloaded on first startup. But when activating the GPU plugin, runelite looks for the contained `.so`s in  `$PWD/natives` for some reason. This is a workaround: download the JARs and extract the `.so`s into the `$out/natives`, then `cd` to `$out` when running runelite.